lib/daru/vector.rb

```diff
@@ -885,37 +885,41 @@ def to_s
 
     # Create a summary of the Vector using Report Builder.
     def summary(method=:to_text)
-      ReportBuilder.new(no_title: true).add(self).send(method)
+      #ReportBuilder.new(no_title: true).add(self).send(method)
+      summary_building
     end
 
     # :nocov:
-    def report_building b # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
-      b.section(name: name) do |s|
-        s.text "n :#{size}"
-        s.text "n valid:#{count_values(*Daru::MISSING_VALUES)}"
-        if @type == :object
-          s.text  "factors: #{factors.to_a.join(',')}"
-          s.text  "mode: #{mode}"
-
-          s.table(name: 'Distribution') do |t|
-            frequencies.sort_by(&:to_s).each do |k,v|
-              key = @index.include?(k) ? @index[k] : k
-              t.row [key, v, ('%0.2f%%' % (v.quo(count_values(*Daru::MISSING_VALUES))*100))]
+    def summary_building level=0 # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
```

Comment:
Well, yes :) But daru was evolving quickly, and now we try to make the code quality better on each PR, without compromises!